overlord/ifacestate/ifacestate_test.go

```diff
@@ -5632,10 +5772,20 @@ func (s *interfaceManagerSuite) TestGadgetConnectConflictRetry(c *C) {
 	c.Assert(tasks, HasLen, 1)
 
 	c.Check(t.Status(), Equals, state.DoingStatus)
-	c.Check(t.Log()[0], Matches, `.*gadget connect will be retried: conflicting snap producer with task "link-snap"`)
+	c.Check(t.Log()[0], Matches, `.*Waiting for conflicting change in progress: conflicting snap producer.*`)
 }
 
-func (s *interfaceManagerSuite) TestGadgetConnectSkipUnknown(c *C) {
+func (s *interfaceManagerSuite) TestAutoConnectGadgetSkipUnknown(c *C) {
+	r := assertstest.MockBuiltinBaseDeclaration([]byte(`
```

Comment:
could you refactor `setupAutoConnectGadget` to allow specifying the gadget yaml to be used in setting up that gadget? Then you could use the gadget.yaml here with the helper and eliminate more of the duplicative code.